### ISSUE TITLE
[18.03] Freeorion : Multiple issues with loading the game.

### ISSUE THREAD

**Issue by cx405** [NixOS/nixpkgs#38615]
## Issue description
Freeorion 0.4.7.1 from 18.03 - loading any saved game does not work.
I use Networkmanager, but it contains only Localhost connection to router, ipv6 is ignored on router.
See below how to replicate the issue and the logs.

```
$ ifconfig 
eno1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.2.142  netmask 255.255.255.0  broadcast 192.168.2.255
        inet6 fe80::3a60:77ff:fe9c:2673  prefixlen 64  scopeid 0x20<link>
        ether 38:60:77:9c:26:73  txqueuelen 1000  (Ethernet)
        RX packets 2097277  bytes 3012043130 (2.8 GiB)
        RX errors 0  dropped 12  overruns 0  frame 0
        TX packets 981510  bytes 93524792 (89.1 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 53073
        device interrupt 20  memory 0xfe700000-fe720000  

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 18594  bytes 2740327 (2.6 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 18594  bytes 2740327 (2.6 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```

### Steps to reproduce
1. Start a new game.
2. Save it.
3. Quit to menu (Resign).
4. Hit Load game.
5. Game loads - the background stars are painted, but then it gets back to main menu.  
A dialog box is drawn: "The connection to the server has been lost." with only choice of [ok].  
In console, this is printed when dialog box appears:
```
terminate called after throwing an instance of 'boost::archive::archive_exception'
  what():  input stream error-Success
```

The `$HOME/.local/share/freeorion/freeorion.log` contains following:
```
<....>
2018-04-08 18:16:58.697935 [debug] Client : HumanClientApp.cpp:600 : HumanClientApp::LoadSinglePlayerGame
2018-04-08 18:16:58.698595 [debug] Client : SaveFileDialog.cpp:671 : pathstring: /home/userp/.local/share/freeorion/save
2018-04-08 18:16:58.699117 [debug] Client : SaveFileDialog.cpp:930 : SaveFileDialog::UpdatePreviewList
2018-04-08 18:16:58.700140 [debug] Client : SaveFileDialog.cpp:597 : SaveFileDialog::FilterColumns: Visible columns: 5
2018-04-08 18:16:58.700621 [debug] Client : SaveFileDialog.cpp:554 : SaveFileDialog::LoadDirectories name: auto valid UTF-8: true
2018-04-08 18:16:58.700814 [debug] Client : SaveGamePreviewUtils.cpp:65 : LoadSaveGamePreviewData: Loading preview from: /home/userp/.local/share/freeorion/save/save-20180408_171459.sav
2018-04-08 18:16:58.700980 [debug] Client : SaveGamePreviewUtils.cpp:82 : Deserializing XML data
2018-04-08 18:16:58.701168 [debug] Client : SaveGamePreviewUtils.cpp:88 : Loaded preview with: 1 human players
2018-04-08 18:16:58.701196 [debug] Client : SaveGamePreviewUtils.cpp:96 : LoadSaveGamePreviewData: Successfully loaded preview from: /home/userp/.local/share/freeorion/save/save-20180408_171459.sav
2018-04-08 18:16:59.645613 [debug] Client : SaveFileDialog.cpp:803 : SaveFileDialog::Confirm: Confirming
2018-04-08 18:16:59.645707 [debug] Client : SaveFileDialog.cpp:819 : choice: save-20180408_171459.sav valid utf-8: true
2018-04-08 18:16:59.645731 [debug] Client : SaveFileDialog.cpp:822 : current dir PathString: /home/userp/.local/share/freeorion/save valid utf-8: true
2018-04-08 18:16:59.645750 [debug] Client : SaveFileDialog.cpp:825 : chosen_full_path PathString: /home/userp/.local/share/freeorion/save/save-20180408_171459.sav valid utf-8: true
2018-04-08 18:16:59.645769 [debug] Client : SaveFileDialog.cpp:826 : chosen_full_path is directory? : false
2018-04-08 18:16:59.661090 [debug] Client : HumanClientApp.cpp:632 : HumanClientApp::LoadSinglePlayerGame() not already in a game, so don't need to end it
2018-04-08 18:16:59.661146 [debug] Client : HumanClientApp.cpp:637 : HumanClientApp::LoadSinglePlayerGame() Starting server
2018-04-08 18:16:59.661168 [debug] Client : HumanClientApp.cpp:377 : HumanClientApp::StartServer: /nix/store/qx0wckc0ip868mh9xvnxi1qzn4v7rnii-freeorion-0.4.7.1/bin/freeoriond
2018-04-08 18:16:59.661194 [debug] Client : HumanClientApp.cpp:407 : ai-config not set.
2018-04-08 18:16:59.662622 [debug] Client : HumanClientApp.cpp:639 : HumanClientApp::LoadSinglePlayerGame() Server started
2018-04-08 18:16:59.662709 [debug] Client : HumanClientApp.cpp:644 : HumanClientApp::LoadSinglePlayerGame() Connecting to server
2018-04-08 18:16:59.662759 [debug] Client : ClientNetworking.cpp:317 : Attempt to connect to server at one of these addresses:
2018-04-08 18:16:59.662816 [debug] Client : ClientNetworking.cpp:319 :   tcp::resolver::iterator host_name: 127.0.0.1  address: 127.0.0.1  port: 12346
2018-04-08 18:16:59.686469 [debug] Client : ClientNetworking.cpp:338 : Connected to server at host_name: 127.0.0.1  address: 127.0.0.1  port: 12346
2018-04-08 18:16:59.686517 [debug] Client : ClientNetworking.cpp:362 : Connecting to server took 23 ms.
2018-04-08 18:16:59.686525 [debug] Client : ClientNetworking.cpp:365 : ConnectToServer() : starting networking thread
2018-04-08 18:16:59.686599 [debug] Client : ClientNetworking.cpp:422 : ClientNetworking::SetPlayerID: player id set to: -1
2018-04-08 18:16:59.686687 [debug] Client : HumanClientFSM.cpp:119 : (HumanClientFSM) IntroMenu.HostSPGameRequested
2018-04-08 18:16:59.686718 [debug] Client : HumanClientFSM.cpp:115 : (HumanClientFSM) ~IntroMenu
2018-04-08 18:16:59.686726 [debug] Client : HumanClientFSM.cpp:147 : (HumanClientFSM) WaitingForSPHostAck
2018-04-08 18:16:59.736631 [debug] Client : HumanClientFSM.cpp:153 : (HumanClientFSM) WaitingForSPHostAck.HostSPGame
2018-04-08 18:16:59.736732 [debug] Client : ClientNetworking.cpp:422 : ClientNetworking::SetPlayerID: player id set to: 1
2018-04-08 18:16:59.738029 [debug] Client : SitRepPanel.cpp:731 : SitRepPanel::Update()
2018-04-08 18:16:59.738312 [debug] Client : DesignWnd.cpp:2847 : DesignWnd::MainPanel::ReregisterDesigns
2018-04-08 18:16:59.738355 [debug] Client : HumanClientFSM.cpp:150 : (HumanClientFSM) ~WaitingForSPHostAck
2018-04-08 18:16:59.738382 [debug] Client : HumanClientFSM.cpp:449 : (HumanClientFSM) PlayingGame
2018-04-08 18:16:59.738425 [debug] Client : HumanClientFSM.cpp:615 : (HumanClientFSM) WaitingForGameStart
2018-04-08 18:16:59.754882 [debug] Client : HumanClientFSM.cpp:589 : (HumanClientFSM) PlayingGame.TurnProgress
2018-04-08 18:17:00.412022 [debug] Client : HumanClientFSM.cpp:589 : (HumanClientFSM) PlayingGame.TurnProgress
2018-04-08 18:17:00.720605 [debug] Client : ClientNetworking.cpp:482 : Client connection disconnected by EOF from server.
2018-04-08 18:17:00.727037 [debug] Client : HumanClientApp.cpp:1201 : HumanClientApp::DisconnectedFromServer
2018-04-08 18:17:00.727080 [debug] Client : HumanClientFSM.cpp:493 : (HumanClientFSM) PlayingGame.Disconnection
2018-04-08 18:17:00.727091 [debug] Client : HumanClientApp.cpp:1154 : HumanClientApp::ResetToIntro
2018-04-08 18:17:00.727099 [debug] Client : HumanClientFSM.cpp:561 : (HumanClientFSM) Quit or reset to main menu.
2018-04-08 18:17:00.727216 [debug] Client : HumanClientFSM.cpp:624 : (HumanClientFSM) ~WaitingForGameStart
2018-04-08 18:17:00.727231 [debug] Client : HumanClientFSM.cpp:457 : (HumanClientFSM) ~PlayingGame
2018-04-08 18:17:00.727239 [debug] Client : HumanClientFSM.cpp:899 : (Host) QuittingGame
2018-04-08 18:17:00.727256 [debug] Client : HumanClientFSM.cpp:906 : (HumanClientFSM) QuittingGame reset to intro is true
2018-04-08 18:17:00.727266 [debug] Client : HumanClientFSM.cpp:916 : (HumanClientFSM) QuittingGame.ShutdownServer
2018-04-08 18:17:00.727279 [debug] Client : HumanClientFSM.cpp:985 : (HumanClientFSM) QuittingGame.TerminateServer
2018-04-08 18:17:00.727288 [debug] Client : HumanClientFSM.cpp:988 : QuittingGame terminated server process.
2018-04-08 18:17:00.727298 [debug] Client : Process.cpp:234 : Process::Impl::Kill calling kill(m_process_id, SIGKILL)
2018-04-08 18:17:00.727312 [debug] Client : Process.cpp:236 : Process::Impl::Kill calling waitpid(m_process_id, &status, 0)
2018-04-08 18:17:00.727332 [debug] Client : Process.cpp:238 : Process::Impl::Kill done
2018-04-08 18:17:00.727343 [debug] Client : ClientNetworking.cpp:422 : ClientNetworking::SetPlayerID: player id set to: -1
2018-04-08 18:17:00.727626 [debug] Client : SitRepPanel.cpp:731 : SitRepPanel::Update()
2018-04-08 18:17:00.727924 [debug] Client : DesignWnd.cpp:2847 : DesignWnd::MainPanel::ReregisterDesigns
2018-04-08 18:17:00.727961 [debug] Client : HumanClientFSM.cpp:903 : (HumanClientFSM) ~QuittingGame
2018-04-08 18:17:00.727974 [debug] Client : HumanClientFSM.cpp:110 : (HumanClientFSM) IntroMenu
```

The `$HOME/.local/share/freeorion/freeoriond.log` contains following:
```
2018-04-08 18:16:59.686552 [debug] Server : Logger.cpp:67 : Logger initialized
2018-04-08 18:16:59.686643 [info] Server : Logger.cpp:68 : v0.4.7.1 [build ???] CMake
2018-04-08 18:16:59.686658 [info] Server : DependencyVersions.cpp:81 : Dependency versions from headers:
2018-04-08 18:16:59.686687 [info] Server : DependencyVersions.cpp:84 : Boost: 1_66
2018-04-08 18:16:59.686709 [info] Server : DependencyVersions.cpp:84 : Python: 2.7.14
2018-04-08 18:16:59.686716 [info] Server : DependencyVersions.cpp:84 : zlib: 1.2.11
2018-04-08 18:16:59.686725 [debug] Server : ServerFSM.cpp:252 : (ServerFSM) Idle
2018-04-08 18:16:59.686739 [debug] Server : ServerApp.cpp:311 : FreeOrion server waiting for network events
2018-04-08 18:16:59.686756 [debug] Server : ServerNetworking.cpp:582 : ServerNetworking::AcceptConnection : connected to new player
2018-04-08 18:16:59.686788 [debug] Server : ServerNetworking.cpp:233 : Server received message from player id: -1 of type Host SP Game and size 955
2018-04-08 18:16:59.686807 [debug] Server : ServerFSM.cpp:290 : (ServerFSM) Idle.HostSPGame
2018-04-08 18:16:59.723377 [debug] Server : SaveLoad.cpp:428 : Reading player save game data from: /home/userp/.local/share/freeorion/save/save-20180408_171459.sav
2018-04-08 18:16:59.723429 [debug] Server : SaveLoad.cpp:437 : Attempting binary deserialization...
2018-04-08 18:16:59.723531 [debug] Server : SaveLoad.cpp:447 : Trying again with XML deserialization...
2018-04-08 18:16:59.723636 [debug] Server : SaveLoad.cpp:459 : Done reading player save game data...
2018-04-08 18:16:59.723656 [debug] Server : ServerNetworking.cpp:121 : PlayerConnection(@ 0x10dba40)::EstablishPlayer(1, Imperor Terran, 1, v0.4.7.1 [build ???] CMake)
2018-04-08 18:16:59.723708 [debug] Server : ServerFSM.cpp:255 : (ServerFSM) ~Idle
2018-04-08 18:16:59.723718 [debug] Server : ServerFSM.cpp:1056 : (ServerFSM) WaitingForSPGameJoiners
2018-04-08 18:16:59.723754 [debug] Server : SaveLoad.cpp:428 : Reading player save game data from: /home/userp/.local/share/freeorion/save/save-20180408_171459.sav
2018-04-08 18:16:59.723776 [debug] Server : SaveLoad.cpp:437 : Attempting binary deserialization...
2018-04-08 18:16:59.723812 [debug] Server : SaveLoad.cpp:447 : Trying again with XML deserialization...
2018-04-08 18:16:59.723902 [debug] Server : SaveLoad.cpp:459 : Done reading player save game data...
2018-04-08 18:16:59.723919 [debug] Server : ServerApp.cpp:188 : ServerApp::CreateAIClients: 4 player (maybe not all AIs) at max aggression: 5
2018-04-08 18:16:59.723955 [debug] Server : ServerApp.cpp:337 : ServerApp::CleanupAIs() telling AIs game is ending
2018-04-08 18:16:59.723966 [debug] Server : ServerApp.cpp:357 : ServerApp::CleanupAIs() killing 0 AI clients.
2018-04-08 18:16:59.723980 [debug] Server : ServerApp.cpp:234 : starting AIs with /nix/store/qx0wckc0ip868mh9xvnxi1qzn4v7rnii-freeorion-0.4.7.1/bin/freeorionca
2018-04-08 18:16:59.723989 [debug] Server : ServerApp.cpp:235 : ai-aggression set to 5
2018-04-08 18:16:59.723996 [debug] Server : ServerApp.cpp:236 : ai-path set to 'python/AI'
2018-04-08 18:16:59.724004 [debug] Server : ServerApp.cpp:243 : ai-config not set.
2018-04-08 18:16:59.724248 [debug] Server : ServerApp.cpp:261 : done starting AI AI_3
2018-04-08 18:16:59.724470 [debug] Server : ServerApp.cpp:261 : done starting AI AI_2
2018-04-08 18:16:59.724633 [debug] Server : ServerApp.cpp:261 : done starting AI AI_1
2018-04-08 18:16:59.724690 [debug] Server : CommonFramework.cpp:38 : Initializing FreeOrion Python interface
2018-04-08 18:16:59.731097 [debug] Server : CommonFramework.cpp:56 : Python initialized
2018-04-08 18:16:59.731144 [debug] Server : CommonFramework.cpp:57 : Python version: 2.7.14 (default, Sep 16 2017, 17:49:51) 
[GCC 7.3.0]
2018-04-08 18:16:59.731157 [debug] Server : CommonFramework.cpp:58 : Python prefix: /nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14
2018-04-08 18:16:59.731166 [debug] Server : CommonFramework.cpp:59 : Python module search path: /nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14/lib/python27.zip:/nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14/lib/python2.7:/nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14/lib/python2.7/plat-linux2:/nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14/lib/python2.7/lib-tk:/nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14/lib/python2.7/lib-old:/nix/store/nx3jw576gqw01iiijgsav39w2qa4cni2-python-2.7.14/lib/python2.7/lib-dynload
2018-04-08 18:16:59.731194 [debug] Server : CommonFramework.cpp:66 : Initializing C++ interfaces for Python
2018-04-08 18:16:59.731473 [debug] Server : ServerFramework.cpp:59 : Initializing server Python modules
2018-04-08 18:16:59.821586 [debug] Server : LoggingWrapper.cpp:23 : Python stdout and stderr redirected
2018-04-08 18:17:00.398600 [debug] Server : ServerFramework.cpp:93 : Server Python modules successfully initialized!
2018-04-08 18:17:00.398641 [debug] Server : CommonFramework.cpp:97 : FreeOrion Python interface successfully initialized!
2018-04-08 18:17:00.398656 [debug] Server : ServerFSM.cpp:1215 : (ServerFSM) WaitingForSPGameJoiners.CheckStartConditions
2018-04-08 18:17:00.398681 [debug] Server : ServerNetworking.cpp:582 : ServerNetworking::AcceptConnection : connected to new player
2018-04-08 18:17:00.398709 [debug] Server : ServerNetworking.cpp:582 : ServerNetworking::AcceptConnection : connected to new player
2018-04-08 18:17:00.398727 [debug] Server : ServerNetworking.cpp:233 : Server received message from player id: -1 of type Join Game and size 317
2018-04-08 18:17:00.398744 [debug] Server : ServerFSM.cpp:1160 : (ServerFSM) WaitingForSPGameJoiners.JoinGame
2018-04-08 18:17:00.398753 [debug] Server : Message.cpp:805 : ExtractJoinGameMessageData() from Default_Player_Name_in_WaitingForSPGameJoiners::react(const JoinGame& msg) client type -1
2018-04-08 18:17:00.398809 [debug] Server : ServerNetworking.cpp:121 : PlayerConnection(@ 0x1171930)::EstablishPlayer(2, AI_3, 0, v0.4.7.1 [build ???] CMake)
2018-04-08 18:17:00.398851 [debug] Server : ServerFSM.cpp:1215 : (ServerFSM) WaitingForSPGameJoiners.CheckStartConditions
2018-04-08 18:17:00.398867 [debug] Server : ServerNetworking.cpp:582 : ServerNetworking::AcceptConnection : connected to new player
2018-04-08 18:17:00.398887 [debug] Server : ServerNetworking.cpp:233 : Server received message from player id: -1 of type Join Game and size 317
2018-04-08 18:17:00.398901 [debug] Server : ServerFSM.cpp:1160 : (ServerFSM) WaitingForSPGameJoiners.JoinGame
2018-04-08 18:17:00.398909 [debug] Server : Message.cpp:805 : ExtractJoinGameMessageData() from Default_Player_Name_in_WaitingForSPGameJoiners::react(const JoinGame& msg) client type -1
2018-04-08 18:17:00.398943 [debug] Server : ServerNetworking.cpp:121 : PlayerConnection(@ 0x1756dd0)::EstablishPlayer(3, AI_2, 0, v0.4.7.1 [build ???] CMake)
2018-04-08 18:17:00.398968 [debug] Server : ServerFSM.cpp:1215 : (ServerFSM) WaitingForSPGameJoiners.CheckStartConditions
2018-04-08 18:17:00.398983 [debug] Server : ServerNetworking.cpp:233 : Server received message from player id: -1 of type Join Game and size 317
2018-04-08 18:17:00.398995 [debug] Server : ServerFSM.cpp:1160 : (ServerFSM) WaitingForSPGameJoiners.JoinGame
2018-04-08 18:17:00.399004 [debug] Server : Message.cpp:805 : ExtractJoinGameMessageData() from Default_Player_Name_in_WaitingForSPGameJoiners::react(const JoinGame& msg) client type -1
2018-04-08 18:17:00.399036 [debug] Server : ServerNetworking.cpp:121 : PlayerConnection(@ 0x1755820)::EstablishPlayer(4, AI_1, 0, v0.4.7.1 [build ???] CMake)
2018-04-08 18:17:00.399059 [debug] Server : ServerFSM.cpp:1215 : (ServerFSM) WaitingForSPGameJoiners.CheckStartConditions
2018-04-08 18:17:00.399069 [debug] Server : ServerFSM.cpp:1220 : WaitingForSPGameJoiners::react(const CheckStartConditions& u) : have all 4 expected players connected.
2018-04-08 18:17:00.399081 [debug] Server : ServerFSM.cpp:1227 : Loading SP game save file: /home/userp/.local/share/freeorion/save/save-20180408_171459.sav
2018-04-08 18:17:00.399205 [debug] Server : SaveLoad.cpp:323 : Allocating buffers for XML deserialization...
2018-04-08 18:17:00.413094 [debug] Server : SerializeUniverse.cpp:70 : Universe::serialize : deserializing universe width: 720.386
2018-04-08 18:17:00.667329 [debug] Server : SerializeUniverse.cpp:75 : Universe::serialize : deserializing 0 ship designs
2018-04-08 18:17:00.669164 [debug] Server : SerializeUniverse.cpp:83 : Universe::serialize : deserializing empire object visibility for 4, 4, 4, 4 empires
2018-04-08 18:17:00.686132 [debug] Server : SerializeUniverse.cpp:96 : Universe::serialize : deserializing 214 objects
2018-04-08 18:17:00.686179 [debug] Server : SerializeUniverse.cpp:102 : Universe::serialize : deserializing 0 destroyed object ids
2018-04-08 18:17:00.697395 [debug] Server : SerializeUniverse.cpp:109 : Universe::serialize : deserializing empire known objects for 4 empires
2018-04-08 18:17:00.697425 [debug] Server : SerializeUniverse.cpp:114 : Universe::serialize : deserializing last allocated ids
2018-04-08 18:17:00.698330 [debug] Server : SerializeUniverse.cpp:118 : Universe::serialize : deserializing 18 types of statistic
2018-04-08 18:17:00.698343 [debug] Server : SerializeUniverse.cpp:120 : Universe::serialize : deserializing done
2018-04-08 18:17:00.698350 [debug] Server : SerializeUniverse.cpp:131 : Universe::serialize : updating empires' latest known object destruction states
2018-04-08 18:17:00.698404 [debug] Server : SerializeUniverse.cpp:140 : Universe::serialize done
```

## Technical details
- system: `"x86_64-linux"`
 - host os: `Linux 4.15.15, NixOS, 18.03.131807.489a14add9a (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixos-18.03.131807.489a14add9a, nixos-master-18.09pre133640.ea145b68a01"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by cx405** [comment]
Pinging contributors of Freeorion for awareness:  @rardiol @abbradar @peterhoeg @Ma27 @Infinisil 
@avnik

**Comment by cx405** [comment]
There is a workaround for the issue, problems stops if in "Options"->"Other", the option "Create Binary Save Files" is ticked (ON).

This reminds me of: https://github.com/freeorion/freeorion/issues/1061  which might indicate problems with boost serialization. Might be Nixos-only. 

The save file in non-binary(default) mode has these fields:
```
<?xml version="1.0" encoding="UTF-8" standalone="yes" ?>
<!DOCTYPE boost_serialization>
<boost_serialization signature="serialization::archive" version="16">
...XML text data here...
<compressed_str>
...!! encrypted binary data here...
</compressed_str>
</boost_serialization>
```

The XML text data block is looking fine, but encrypted data could be corrupted.

**Comment by cx405** [comment]
Now I get hit by "save game in progress" after just 3 turns. Looks like it didn't make into the game, so disabling buggy autosave would fix it for now.

https://github.com/freeorion/freeorion/issues/1405

**Comment by cx405** [comment]
This is crazy.. After disabling the autosave and playing through 3 turns, I am thrown back at Menu with the same message as in OP "Disconnected from server", and I am thrown back into menu if I try to load the game.

However, I can quit freeorion and then load the game.

The console spills the same error message, when I am thrown back to menu:
```
terminate called after throwing an instance of 'boost::archive::archive_exception'
  what():  input stream error-Resource temporarily unavailable

```

**Comment by cx405** [comment]
This could be related: https://github.com/freeorion/freeorion/issues/1898

**Comment by Ma27** [comment]
Although I've been pinged I'm not sure if I can actually help here - I just applied a simple upstream patch to fix the build with `libboost` before the 18.03 release, so I don't have sufficient knowledge regarding `freeorion`'s internals, sorry!

**Comment by aanderse** [comment]
@cx405 looks like this has been fixed. Unless you're still having this issue can you please close?
